### DISCUSSION TITLE
Limited number of additional parts from a pool?

### DISCUSSION THREAD

**Discussion by Kiaazad** [nftchef/art-engine#22] [Help + Q&A] [answered]
I have 3 main parts present in all images (4 including a background) and ~20 optional parts, is it possible to give each image a random number of optional parts limited to 5?

**Comment by nftchef** [comment, marked as answer]
Yes, 
In any/all of the folders, you can add a `NONE.png` empty/transparent PNG image with a `#` weight. if chosen, that folders trait will be skipped and not included in the metadata

example,
![image](https://user-images.githubusercontent.com/91582112/152701961-458a7da5-dd70-45ce-beb8-e33b4eb5d946.png)

**Comment by Kiaazad** [comment]
I assumed: that would give me possibility of one image getting up to 20 optional parts.
Is there any way to make sure it's limited to 5?

**Comment by nftchef** [comment]
Any folder with any given number of files with a `#` weght will only choose `1` of those files. 
To do multiple optional traits, you would need to have multiple folders for each trait, using `#` weights and `NONE` within each folder.

In the following example, every `OptionalTrait` folder will "roll" for a single file with that folder, with the percent chance of each file being, 

`weight/sum of weights in folder`

in the examples below, each trait has,
50% chance of being NONE, or not being selected,
25% Option C
12.5% Option A
12.5% Option B

![image](https://user-images.githubusercontent.com/91582112/152808692-b5893bc7-052a-4f89-bbf9-f5e74375a9db.png)

**Reply by sktchapp** [reply]
In this example, is it possible to bundle the multiple optionaltraits under a single trait_type category in the metadata? Having the root folder as Accessory (or trait: "Accessory") so that you get 3 different values in the json under Accessory as opposed to them overwriting each other if all three traits are chosen?
